### ISSUE TITLE
Add dark/night theme or an "invert colors" mode

### ISSUE THREAD

**Issue by gussmith23** [aardappel/treesheets#107]
This may be a pretty niche feature request. It may also be unnecessary due to the existing ability to change colors in TreeSheets. I will explain my reasoning nonetheless.

As someone who suffers from eye strain, I generally seek to have dark themes on all of my programs. I used to achieve this using light themes plus a system-wide color inverter; this is not preferable, though, for a number of reasons. Now, I'm generally able to achieve an all-dark-themed system with the high contrast Chrome extension plus dark themes in all of my programs. Luckily, dark themes have become more and more popular, so this works 99% of the time.

TreeSheets is one of the exceptions unfortunately. I understand that a TreeSheet has a color palette, and that the default color palette is black text on white. I believe this default is reasonable, and I don't want to change it. What would be nice would be the option to invert this color scheme _locally_ within TreeSheets, so that on my local machine, I could view a black-on-white sheet as white-on-black. This was the cleanest way I could think of to solve this problem, as it doesn't involve changing a .cts file to fit my color preferences.

**Comment by aardappel** [comment]
Hmm.. not sure what it would look like if you simply inverted everything. At the least it would need to be gamma corrected to look decent I think.

Could do this on a bitmap level which would probably be too slow. The better way may to simply invert any color when one is set, and hope it renders correctly :)

Would certainly be a bit of work to make sure everything works in such an inverted mode.

The current workaround of course is to set the colors of the sheet as you say, but that is annoying if you cooperate with others.

**Comment by VegaDeftwing** [comment]
How does the current theme work? I know we can change the default document background color, but if we could change the default cell, border, and text colors that gets this issue pretty close at least. It still leaves the problem of the top tool bar of course, assuming that it isn't provided by QT (I don't think it's GTK, but I also haven't looked into the project much yet) or otherwise it may be easy enough just to add a dark mode for that as well as the icons look like they'd support a dark theme well enough

**Comment by aardappel** [comment]
Yup, you can change all of them, check out the top toolbar.

**Comment by VegaDeftwing** [comment]
Maybe for faster and 'prettier' theme support something like https://github.com/hundredrabbits/Themes could be added? It's basically just a color pallet override

**Comment by skfjirim** [comment]
I just found this great app, and when I finnished the tutorial, only thing what I was looking for was  "dark mode" button.. :(

**Comment by gerroon** [comment]
Are there any plans to implement a fully dark theme. It is very painful for some people to look at bright colors on the screens. 

A dark theme should not be just dark background, the whole window needs to support it because it is even more painful to look at something dark that is framed with something very bright (like the current frame colors)

thanks

**Comment by aardappel** [comment]
Yup, aware just changing the sheet is only a workaround.. it be a fair bit of work to do a full dark mode.

**Comment by thadeudepaula** [comment]
May be if it only respects the system colors. Today browsers understand the "night mode" from OS. Sometimes dark themes may be buggy, and the user could simply choose between the OS color or the white default. It would bring backward compatibility with old files that would look buggy in dark envs (as red or blue text on dark background).

If there was a way, perhaps a cts color mapper... so people could map the used colors to a predefined16 or 32 colors. These way community could share colorschemes in simple pair/value formats. Once a file has the map used the exports would be fine too.

**Comment by aardappel** [comment]
Sounds like dark mode in wxWidgets (what TreeSheets uses for the UI) is still very much a work in progress, and only works on the mac so far: https://forums.wxwidgets.org/viewtopic.php?t=45869 https://forums.wxwidgets.org/viewtopic.php?t=47110

**Comment by huangfeiqiannian** [comment]
there is an autohotkey way to get your treesheets into black mode.

see:
https://www.autohotkey.com/boards/viewtopic.php?t=94011 

but its not easy for nomal users.

the result:
```
   Matrix := "-0.8|0|0|0|0|"
           . "0|-0.8|0|0|0|"
           . "0|0|-0.8|0|0|"
           . "0|0|0|1|0|"
           . "1|1|1|0|1"
```
![image](https://user-images.githubusercontent.com/42437184/181870969-a17ce6ad-fc73-4c89-88ee-c46108565cdf.png)

or total dark (not recommend)
```
   Matrix := "-1|0|0|0|0|"
           . "0|-1|0|0|0|"
           . "0|0|-1|0|0|"
           . "0|0|0|1|0|"
           . "1|1|1|0|1"
```
![image](https://user-images.githubusercontent.com/42437184/181871006-8fbf6c68-ac51-4f41-b18c-441542d15e0f.png)

**Comment by vinniec2** [comment]
I stopped by because I just discovered this program and was curious to try it out (it reminds me a lot of the GUI https://eaglemode.sourceforge.net). I opened it and was already looking for the dark theme (yes, I'm one of those people with tired eyes).
For participants who use Linux, when Compiz was around, you could easily set the negative of a single window, then Compiz disappeared. Currently, as far as I know, this feature is available in the basic KDE (I tried it and it works), and I read that the composite manager “compton” also integrated it, if I remember correctly.

**Comment by jlane003** [comment]
Now that dark mode is available in wxWidgets, are there any plans on bringing it to treesheets?
